samples/BookStore-Angular-MongoDb/aspnet-core/src/Acme.BookStore.HttpApi.Host/BookStoreHttpApiHostModule.cs

```diff
@@ -153,16 +153,15 @@ public override void OnApplicationInitialization(ApplicationInitializationContex
             app.UseVirtualFiles();
             app.UseRouting();
             app.UseCors(DefaultCorsPolicyName);
-            app.UseAuthentication();
-            app.UseAuthorization();
+            app.UseIdentityServer();
             app.UseJwtTokenMiddleware();
 
             if (MultiTenancyConsts.IsEnabled)
             {
                 app.UseMultiTenancy();
             }
 
-            app.UseIdentityServer();
+            app.UseAuthorization();
```

Comment:
We can not do this, because;

* `UseMultiTenancy` requires `UseAuthentication` comes before it, because it gets tenantid from the current claims.
* `UseIdentityServer` can not come before `UseMultiTenancy`, otherwise identity server can not run multi-tenant.

So, the only solution is to duplicate authorization middleware (one added by us, one added by IDS4).

Reply:
Another problem is that `UseJwtTokenMiddleware `needs to be before `UseAuthorization`. But it is now fixed by Https://github.com/abpframework/abp/commit/f767232d42cc7076b87d5f52911b02eb15b9fc44.